src/SamplesApp/UITests.Shared/Windows_Graphics_Display/DisplayInformationTests.xaml.cs

```diff
@@ -0,0 +1,167 @@
+				RaisePropertyChanged();
+			}
+		}
+
+		public string LastDpiChangeInfo { get; private set; }
+
+		public string LastOrientationChangeInfo { get; private set; }
+
+		public ObservableCollection<PropertyInformation> Properties { get; set; } = new ObservableCollection<PropertyInformation>();
+
+		public ICommand ObserveDpiChangesCommand => GetOrCreateCommand(ObserveDpiChanges);
+
+		private void ObserveDpiChanges()
+		{
+			var info = DisplayInfo.GetForCurrentView();
+			if (DpiChangesOn)
+			{
+				info.DpiChanged += DpiChanged;
+			}
+			else
+			{
+				info.DpiChanged -= DpiChanged;
+			}
+		}
+
+		public ICommand ObserveOrientationChangesCommand => GetOrCreateCommand(ObserveOrientationChanges);
+
+		private void ObserveOrientationChanges()
+		{
+			var info = DisplayInfo.GetForCurrentView();
+			if (OrientationChangesOn)
+			{
+				info.OrientationChanged += OrientationChanged;
+			}
+			else
+			{
+				info.OrientationChanged -= OrientationChanged;
+			}
+		}
+
+		private void DpiChanged(DisplayInfo sender, object args)
+		{
+			LastDpiChangeInfo = $"{DateTime.Now.ToLongTimeString()} ({sender.LogicalDpi})";
+			RaisePropertyChanged(nameof(LastDpiChangeInfo));
+		}
+
+		private void OrientationChanged(DisplayInfo sender, object args)
+		{
+			LastOrientationChangeInfo = $"{DateTime.Now.ToLongTimeString()} ({sender.CurrentOrientation})";
+			RaisePropertyChanged(nameof(LastOrientationChangeInfo));
+		}
+
+		private void RefreshDisplayInformation()
+		{			
+			var info = DisplayInfo.GetForCurrentView();
+			var properties = new PropertyInformation[]
+			{
+				new PropertyInformation(nameof(DisplayInfo.AutoRotationPreferences), SafeGetValue(()=>DisplayInfo.AutoRotationPreferences)),
+				new PropertyInformation(nameof(info.CurrentOrientation), SafeGetValue(()=>info.CurrentOrientation)),
+				new PropertyInformation(nameof(info.NativeOrientation), SafeGetValue(()=>info.NativeOrientation)),
+				new PropertyInformation(nameof(info.ScreenHeightInRawPixels), SafeGetValue(()=>info.ScreenHeightInRawPixels)),
+				new PropertyInformation(nameof(info.ScreenWidthInRawPixels), SafeGetValue(()=>info.ScreenWidthInRawPixels)),
+				new PropertyInformation(nameof(info.LogicalDpi), SafeGetValue(()=>info.LogicalDpi)),
+				new PropertyInformation(nameof(info.DiagonalSizeInInches), SafeGetValue(()=>info.DiagonalSizeInInches)),
+				new PropertyInformation(nameof(info.RawPixelsPerViewPixel), SafeGetValue(()=>info.RawPixelsPerViewPixel)),
+				new PropertyInformation(nameof(info.RawDpiX), SafeGetValue(()=>info.RawDpiX)),
+				new PropertyInformation(nameof(info.RawDpiY), SafeGetValue(()=>info.RawDpiY)),
+				new PropertyInformation(nameof(info.ResolutionScale), SafeGetValue(()=>info.ResolutionScale)),
+			};
+			Properties = new ObservableCollection<PropertyInformation>(properties);
+			RaisePropertyChanged(nameof(Properties));
+		}
+
+		private string SafeGetValue<T>(Func<T> getter)
+		{
+			try
+			{
+				var value = getter();
+				if (value == null)
```

Comment:
Codacy found an issue: [Use a comparison to 'default(T)' instead or add a constraint to 'T' so that it can't be a value type.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5885845)